stackrox/stackrox
# ROX-23245: add acscsEmail notifier boilerplate

## Description

This PR adds the initial boilerplate to register a new acscsEmail notifier integration type.

There's no change in the notifier.proto because the notifier will only need the common fields:
- Name
- LabelKey
- LabelDefault

I'll follow up with the actual implementation in other PRs, which is why I have guarded the registration of this notifier with a feature flag for now. 


## Checklist

If any of these don't apply, please comment below.

## Testing Performed

- Local build of central
- First try to create a notifier which should fail:
```
curl -v 'https://admin:ldPXycggtb36mLU09Oo2QoBPP@localhost:8000/v1/notifiers' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'content-type: application/json' \
  --data-raw '{"id":"","name":"test","labelDefault":"test@testitest","labelKey":"testitest","uiEndpoint":"https://localhost:8000","type":"acscsEmail"}' \
  --insecure
```
- Restart with proper config env vars `ROX_ACSCS_EMAIL_NOTIFIER=true` and `ROX_MANAGED_CENTRAL=true`
- Use above command again, should succeed now

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.
